sdk/servicebus/service-bus/src/serviceBusAtomManagementClient.ts

```diff
@@ -361,30 +367,72 @@ export class ServiceBusManagementClient extends ServiceClient {
   private ruleResourceSerializer: AtomXmlSerializer;
 
   /**
-   * SAS token provider used to generate tokens as required for the various operations.
+   * Token provider used to generate tokens as required for the various operations.
    */
-  private sasTokenProvider: SharedKeyCredential;
+  private tokenProvider: SharedKeyCredential | TokenCredential;
 
   /**
    * Initializes a new instance of the ServiceBusManagementClient class.
    * @param connectionString The connection string needed for the client to connect to Azure.
    * @param options ServiceBusManagementClientOptions
    */
-  constructor(connectionString: string, options?: ServiceBusManagementClientOptions) {
-    const connectionStringObj: any = parseConnectionString(connectionString);
-
-    if (connectionStringObj.Endpoint == undefined) {
-      throw new Error("Missing Endpoint in connection string.");
-    }
-
-    const credentials = new SasServiceClientCredentials(
-      connectionStringObj.SharedAccessKeyName,
-      connectionStringObj.SharedAccessKey
-    );
-
+  constructor(connectionString: string, options?: ServiceBusManagementClientOptions);
+  /**
+   *
+   * @param fullyQualifiedNamespace The fully qualified namespace of your Service Bus instance which is
+   * likely to be similar to <yournamespace>.servicebus.windows.net.
+   * @param credential A credential object used by the client to get the token to authenticate the connection
+   * with the Azure Service Bus. See &commat;azure/identity for creating the credentials.
+   * If you're using your own implementation of the `TokenCredential` interface against AAD, then set the "scopes" for service-bus
+   * to be `["https://servicebus.azure.net//user_impersonation"]` to get the appropriate token.
+   * @param options ServiceBusManagementClientOptions
+   */
+  constructor(
+    fullyQualifiedNamespace: string,
+    credential: TokenCredential,
+    options?: ServiceBusManagementClientOptions
+  );
+
+  constructor(
+    fullyQualifiedNamespaceOrConnectionString1: string,
+    credentialOrOptions2?: TokenCredential | ServiceBusManagementClientOptions,
+    options3?: ServiceBusManagementClientOptions
+  ) {
     const requestPolicyFactories: RequestPolicyFactory[] = [];
-    requestPolicyFactories.push(signingPolicy(credentials));
-
+    let options: ServiceBusManagementClientOptions;
+    let fullyQualifiedNamespace: string;
+    let credentials: SasServiceClientCredentials | TokenCredential;
+    let tokenProvider: SharedKeyCredential | TokenCredential;
```

Comment:
For a future PR, can we consider not having "tokenProvider" and make do with just the "credentials"?
We can update `SasServiceClientCredentials` to internally create a `SharedKeyCredential` and expose its `getToken()` method.

Reply:
I'll try that!

Reply:
Done!